Application/src/model/Menu.java

```diff
@@ -209,25 +244,151 @@ private void setEvents(Stage stage) {
+                        } catch (NullPointerException e) {
+                            continue;
+                        }
+
+                        // Catches if there is no parallel diagram.
+                        try {
+                            Net.push("{share, " + parse.getParallelSequenceDiagram() + "}");
+                        } catch (NullPointerException e) {
+                            continue;
+                        }
+
+                        break;
+                    case "class_diagram":
+                        parse.parseClassDiagram(file);
+                        System.out.println(parse.getDiagram()); // TODO print until backend is implemented.
+                        Net.push("{share, " + parse.getDiagram() + "}");
+
+                        break;
+                }
+
+            }
+            // if the diagram is not included in the switch case, check if the diagram is invalid
+            DiagramCheck.ContainsDiagram(result);
+            identifyState();
+        });
+
         button_previous.setOnAction((ActionEvent event) ->{
-            Net.push("{" + DiagramView.getDiagramViewInView().getTab().getId() + ", previous_message}");
+            String tabID = DiagramView.getDiagramViewInView().getTab().getId();
+            try {
+                //Checks if the current tab has a diagram from a lobby or not
+                if (tabID.charAt(1) == 'l') {
+                    Net.push("{share, {" + tabID + ", previous_message}}");
+                } else {
+                    Net.push("{" + tabID + ", previous_message}");
+                }
+            }catch(StringIndexOutOfBoundsException e){
+                Net.push("{" + tabID + ", previous_message}");
+            }
         });
 
         button_next.setOnAction((ActionEvent event)    ->{
-            Net.push("{" + DiagramView.getDiagramViewInView().getTab().getId() + ", next_message}");
+            String tabID = DiagramView.getDiagramViewInView().getTab().getId();
+            try {
+                //Checks if the current tab has a diagram from a lobby or not
+                if (tabID.charAt(1) == 'l') {
+                    Net.push("{share, {" + tabID + ", next_message}}");
+                } else {
+                    Net.push("{" + tabID + ", next_message}");
+                }
+            }catch(StringIndexOutOfBoundsException e){
+                Net.push("{" + tabID + ", next_message}");
+            }
         });
 
         button_auto.setOnAction((ActionEvent event)    ->{
             play = button_auto.getText().endsWith(text_auto_play);
             identifyState();
         });
 
+        button_create_lobby.setOnAction((ActionEvent event)    ->{
+            passwordBox();
+            identifyState();
+        });
+        button_remove_lobby.setOnAction((ActionEvent event) ->{
+            Net.push("{" + "share, " + "remove_lobby}");
+
+            host = false;
```

Comment:
Now you are assuming that you can do this?

Reply:
?